docs/syntax/Cadence.syntax

```diff
@@ -0,0 +1,1482 @@
+    ;
+
+<decimalToken>:
+        "0"
+    |   "123456789" *"0123456789_" ;
+
+<binaryToken>:
+        "0" "b" 1*"01_" ;
+
+<octalToken>:
+        "0" "o" 1*"01234567_" ;
+
+<hexadecimalToken>:
+        "0" "x" 1*"0123456789ABCDEFabcdef_" ;
+
+<invalidNumberToken>:
+        "0" "acdefghijklmnpqrstuvwyzABCDEFGHIJKLMNOPQRSTUVWXYZ" "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ_"
+    ;
+
+<stringToken>:
+        "\"" *quotedTextItem "\"" ;
+
+quotedTextItem:
+        quotedTextItemCharacter
+    |   escapedCharacter
+    ;
+
+quotedTextItemCharacter:
+        ~nonquotedTextItemCharacter
+    ;
+
+nonquotedTextItemCharacter:
+        "\"\r\n\\" ;
+
+escapedCharacter:
+        escapedUnicodeScalar
+    |   escapedCCharacter
+    ;
+
+escapedUnicodeScalar:
+        "\\" "u" "{" 1*8unicodeScalarDigit "}" ;
+
+unicodeScalarDigit:
+        "0123456789abcdefABCDEF" ;
+
+escapedCCharacter:
+        "\\" "0\\tnr\"'" ;
+
+%parser
+
+*
+program:
+         *declaration( declarationSeparator ) $
+    ;
+
+declarationSeparator:
+        semicolon
+    |   empty
+    ;
+
+empty:
+    ;
+
+semicolon:
+        ";"
+    ;
+
+declaration:
+        pragmaDeclaration
+    |   variableDeclaration
+    |   functionDeclaration
+    |   importDeclaration
+    |   eventDeclaration
+    |   structCompositeDeclaration
+    |   resourceCompositeDeclaration
+    |   contractCompositeDeclaration
+    |   enumCompositeDeclaration
+    |   structInterfaceDeclaration
+    |   resourceInterfaceDeclaration
+    |   contractInterfaceDeclaration
+    |   enumInterfaceDeclaration
+    |   entitlementDeclaration
+    |   entitlementMappingDeclaration
+    |   attachmentDeclaration
+    |   transactionDeclaration
+    ;
+
+attachmentDeclaration:
+        access attachmentKeyword nonReservedIdentifier forKeyword nominalType conformances "{" membersAndNestedDeclarations "}"
+    ;
+
+transactionDeclaration:
+        transactionKeyword *1parameterList "{" fields *1prepareFunctionDeclaration *1preConditions *1executeOrPostConditions "}"
+    ;
+
+executeOrPostConditions:
+        executeOrPostConditionsExecute
+    |   executeOrPostConditionsExecutePostConditions
+    |   executeOrPostConditionsPostConditions
+    |   executeOrPostConditionsPostConditionsExecute
+    ;
+
+executeOrPostConditionsExecute:
+        execute
+    ;
+
+executeOrPostConditionsExecutePostConditions:
+        execute postConditions
+    ;
+
+executeOrPostConditionsPostConditions:
+        postConditions
+    ;
+
+executeOrPostConditionsPostConditionsExecute:
+        postConditions execute
+    ;
+
+execute:
+        identifier block
+    ;
+
+importDeclaration:
+        importKeyword *1importDeclarationFromPhrase importDeclarationSource
+    ;
+
+importDeclarationFromPhrase:
+        1*identifier( "," ) fromKeyword
+    ;
+
+importDeclarationSource:
+        stringLiteral
+    |   hexadecimalLiteral
+    |   identifier
+    ;
+
+access:
+        accessPriv
+    |   accessPub
```

Comment:
`pub`, `pub(set)`, and `priv` got removed in Cadence 1.0

Reply:
This syntax is focused on duplicating the behavior of the interpreter's parser.  These would produce errors in the semantic checks. That should also be part of the specification.

Reply:
Maybe at least leave a comment documenting this decision, e.g.

`# Invalid since Cadence 1.0, so not part of the strict syntax`